File: Assignment3/program3.c
////////////////////////////////////////////////////////////////
//
//  File Name   : program3.c
//  Description : Accepts a Integer and Prints it's Even Factorials
//  Author      : Akhilesh.P.Sonavane
//  Date        : 10/05/2025
//
////////////////////////////////////////////////////////////////

#include<stdio.h>

////////////////////////////////////////////////////////////////
// 
//  Function Name : DisplayEvenFactor
//  Description   : Prints Even Factorials of Input Integer
//  Input         : Integer
//  Output        : Integer
//
////////////////////////////////////////////////////////////////

void DisplayEvenFactor(int iNo)
{
    int iCnt = 0;

    if(iNo <= 0)
    {
        iNo = -iNo;
    }
    else 
    {
        for(iCnt = 2; iCnt < iNo; iCnt++)
        {
            if((iNo % iCnt == 0) && (iCnt % 2 == 0)) //If we want both divisible by 2 & 3 just add (iCnt % 3 == 0);
            {
               printf("%d\t",iCnt);
            }    
        }
    }
}
int main()
{
    int iValue = 0;

    printf("Enter Number\n");
    scanf("%d",&iValue);

    DisplayEvenFactor(iValue);
    return 0;
}

File: Assignment23/program3.java
////////////////////////////////////////////////////////////////
//
//  File Name   : program3.java
//  Description : Accepts a Character and displays incremented format for UpperCase Characters
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 29/05/2025
//
////////////////////////////////////////////////////////////////

import java.util.Scanner;

////////////////////////////////////////////////////////////////
// 
//  Class Name    : Display
//  Function Name : Conversion
//  Description   : Inputs a Character and  displays incremented format for UpperCase Characters
//  Input         : Character
//  Output        : Character
//
////////////////////////////////////////////////////////////////

class Display
{
    public void Conversion(char cVal)
    {
        int iCnt = 0;

        if((cVal >= 'A') && (cVal <= 'Z'))
        {
            while(cVal <= 'Z')
            {
                System.out.println(" "+cVal);
                cVal++;
            }
        }
        
        else if((cVal >= 'a') && (cVal <= 'z'))
        {
            while(cVal >= 'a')
            {
                System.out.println(" "+cVal);
                cVal--;
            }
        }
        else 
        {
            return;
        }
    }
}

public class program3
{
    public static void main(String args[])
    {
        char cValue = '\0';
        
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Character : ");
        cValue = sobj.next().charAt(0);

        Display dobj = new Display();

        dobj.Conversion(cValue);

        sobj.close();
    }
} 
